addons/sale/models/sale_order.py

```diff
@@ -849,31 +849,40 @@ def _show_cancel_wizard(self):
         return False
 
     def _find_mail_template(self, force_confirmation_template=False):
-        template_id = False
+        """ Get the appropriate mail template for the current sale's order(s) based on their state.
 
-        if force_confirmation_template or (self.state == 'sale' and not self.env.context.get('proforma', False)):
-            template_id = int(self.env['ir.config_parameter'].sudo().get_param('sale.default_confirmation_template'))
-            template_id = self.env['mail.template'].search([('id', '=', template_id)]).id
-            if not template_id:
-                template_id = self.env['ir.model.data'].xmlid_to_res_id('sale.mail_template_sale_confirmation', raise_if_not_found=False)
-        if not template_id:
-            template_id = self.env['ir.model.data'].xmlid_to_res_id('sale.email_template_edi_sale', raise_if_not_found=False)
+        If all SOs are 'sale' or there is the ``force_confirmation_template``,
+        the mail template is that of the sale confirmation.
+        Else, it returns the quotation's email template.
 
-        return template_id
+        :param bool force_confirmation_template: Allows to force the confirmation template to be
+        used in any case
+        :return: The correct mail template based on the current status
+        :rtype: record of `mail.template`
+        """
+        if force_confirmation_template or (
+            all(order.state == 'sale' for order in self) and not self.env.context.get('proforma')
+        ):
+            _sudo_default_template_id = int(
+                self.env['ir.config_parameter'].sudo()
+                                               .get_param('sale.default_confirmation_template')
+            )
+            _sudo_default_template = self.env['mail.template'].browse(_sudo_default_template_id)
```

Comment:
Since it is a template id, it's an integer, and therefore doesn't have any notion of sudo or not ;)
```suggestion
            default_template_id = int(
                self.env['ir.config_parameter'].sudo()
                                               .get_param('sale.default_confirmation_template')
            )
            default_template = self.env['mail.template'].browse(default_template_id)
```
Furthermore, your _sudo_default_template is not sudoed, since you're only browsing an id, it's confusing to include a sudo in its name.

Reply:
My bad, I saw "sudo" and didn't put more thought in it. I'll keep it in mind for next time.